lib/supavisor/db_handler.ex

```diff
@@ -255,4 +263,83 @@ defmodule Supavisor.DbHandler do
 
     :keep_state_and_data
   end
+
+  # Adapted from Postgrex.Protocol
+  defp handshake(sock, %{upstream_ssl: true} = auth), do: ssl(sock, auth)
+  defp handshake(sock, auth), do: send_startup(sock, auth)
+
+  defp ssl(sock, auth) do
+    case sock_send(sock, Server.ssl_request()) do
+      :ok -> ssl_recv(sock, auth)
+      {:disconnect, _, _} = dis -> dis
+    end
+  end
+
+  @spec ssl_recv(term(), map) ::
+          {:ok, {:ssl, :sslsock}}
+          | {:error, :ssl_not_available | :ssl_connect_error}
+  defp ssl_recv({:gen_tcp, sock} = s, auth) do
+    case :gen_tcp.recv(sock, 1, :infinity) do
```

Comment:
double-checking if infinity is safe here or instead we should, say, 15s? I think that's default timeout for stuff like this that Postgrex uses.

Reply:
no, using infinity is not safe, thank you for the suggestion 👍